iati/resources.py

```diff
@@ -372,11 +464,52 @@ def resource_filesystem_path(path):
     Returns:
         str: A reference to the specified file that works however the package is distributed.
 
+    Raises:
+        TypeError: If the given path is of a type that cannot be a filepath.
+        ValueError: If the given path is a string that cannot be a useful component of a filepath.
+
     Note:
         Does not check to see that the specified file exists.
 
-    Warning:
-        When other functions in this module are reviewed, this will be too.
-
     """
+    try:
+        _ensure_portable_filepath(path)
+    except ValueError:
+        if path != '':
+            raise
+
     return pkg_resources.resource_filename(PACKAGE, path)
+
+
+def _ensure_portable_filepath(maybe_filepath):
+    """Determine whether a string could be a portable filepath.
+
+    A portable filepath is one that meets the POSIX fully portable filename character restrictions:
+
+    * Consists only of: A-Z a-z 0-9 . _ -
+    * Does not have a leading hyphen.
+
+    Args:
+        maybe_filepath (str): A string that might be considered a valid filepath.
+
+    Raises:
+        TypeError: If the input value is of a type that cannot be a filepath.
+        ValueError: If the input value is a string that cannot be a useful component of a filepath.
+
```

Comment:
Add a `Note:` that this restriction could be relaxed, and is present to act as a tight limit to work from (it's easier to relax limits than tighten them!)

Reply:
df072245fb739ce3bf2a967dca53c2c5254d58bb